robotbase/src/main/java/com/systemmeltdown/robot/commands/RecountCellsCommand.java

```diff
@@ -16,41 +14,63 @@
  * 
  * @category Shuffleboard
  */
-public class RecountCellsCommand extends SequentialCommandGroup {
-    private int m_cellCount;
+public class RecountCellsCommand extends CommandBase {
+    public enum StorageState {
+        onCell, betweenCells
+    }
+
     private StorageSubsystem m_storageSubsystem;
+    private FeederSubsystem m_feederSub;
+    private int m_cellCount = 0;
+    private int m_numOfPasses = 0;
+    private StorageState m_storageState;
 
     /**
      * @param storageSub The {@link StorageSubsystem}.
      */
     public RecountCellsCommand(StorageSubsystem storageSubsystem, FeederSubsystem feederSubsystem) {
         m_storageSubsystem = storageSubsystem;
+        m_feederSub = feederSubsystem;
+        addRequirements(storageSubsystem, feederSubsystem);
+    }
 
-        Command nextCell = new RotateStorageSingleCell(m_storageSubsystem);
-        Command countCell = new ConditionalCommand(
-            new InstantCommand(() -> ++m_cellCount),
-            new InstantCommand(() -> {}),
-            feederSubsystem::isFeedSensorBlocked);
-
-        // NOTE: The command group might throw an error because we're reusing these commands
-        // If that happens just move the command creation into the loop.
-        for(int ii = 0; ii < Constants.STORAGE_CAROUSEL_SEGMENTS; ++ii) {
-            addCommands(nextCell, countCell);
+    @Override
+    public void initialize() {
+        if (m_feederSub.isFeedSensorBlocked()) {
+            m_storageState = StorageState.onCell;
+        } else {
+            m_storageState = StorageState.betweenCells;
         }
     }
 
     @Override
-    public void initialize() {
-        super.initialize();
-        m_cellCount = 0;
+    public void execute() {
+        m_storageSubsystem.setRotationSpeed(Constants.STORAGE_CAROUSEL_ROTATION_SPEED);
+        if (m_feederSub.isFeedSensorBlocked()) {
+            if (m_storageState == StorageState.betweenCells) {
+                m_storageState = StorageState.onCell;
+            }
+        } else {
+            if (m_storageState == StorageState.onCell) {
+                m_storageState = StorageState.betweenCells;
+                m_cellCount++;
+            }
+        }
+        // if (m_storageSubsystem.isAlignedForShooting()) {
+        //     m_numOfPasses++;
+        // }
     }
 
     @Override
     public void end(boolean interrupted) {
-        super.end(interrupted);
         m_storageSubsystem.setNumOfCells(m_cellCount);
     }
 
+    @Override
+    public boolean isFinished() {
+        return (m_numOfPasses >= 5);
```

Comment:
I don't see anything incrementing m_numOfPasses in this command.  Are we suppose to be checking the number of cells instead?